core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -1955,6 +1957,263 @@ public static <A, ResultContainer_, Result_> UniConstraintCollector<A, ResultCon
         return InnerQuadConstraintCollectors.toConsecutiveSequences(resultMap, indexMap);
     }
 
+    // *****************************************************************
+    // concurrentUsage
+    // *****************************************************************
+    /**
+     * Creates a constraint collector that returns {@link ConcurrentUsageInfo} about the first fact.
+     *
+     * For instance, {@code [Equipment from=2, to=4] [Equipment from=3, to=5] [Equipment from=6, to=7] [Equipment from=7, to=8]}
+     * returns the following information:
+     *
+     * <pre>
+     * {@code
+     * ConcurrentUsage: [minConcurrentUsage: 1, maxConcurrentUsage: 2,
+     *                  [Equipment from=2, to=4] [Equipment from=3, to=5]],
+     *                  [minConcurrentUsage: 1, maxConcurrentUsage: 1,
+     *                  [Equipment from=6, to=7] [Equipment from=7, to=8]]
```

Comment:
Looking at the example, shouldn't the `to` rather be exclusive?
Java APIs concerning ranges typically follow conventions `[min, max)`.

Reply:
(in particular, putting `[6, 7)` and `[7, 8)` into two seperate clusters means creating a break `[7, 7)`, which has length `0`)

Reply:
But isn't that correct?
`[6, 7)` and `[7, 8)` aren't overlapping, so there _should_ be a break between them. (Even though the length is technically zero.)

What happens with `[6, 7]` and `[7, 8]` - is that an overlap?
And how do you model a situation where you don't _want_ an overlap in that situation?

IMO we need to be very clear about the conceptual model of this thing; no matter what it is.